src/software/ai/evaluation/indirect_chip.cpp

```diff
@@ -177,62 +175,21 @@ std::vector<LegacyTriangle> Evaluation::findOpenTriangles(
     return filtered_triangles;
 }
 
-Point Evaluation::getTriangleCenter(LegacyTriangle triangle)
+std::vector<Triangle> Evaluation::removeTrianglesOutsideRectangle(
+    Rectangle rectangle, std::vector<Triangle> triangles)
 {
-    Point p1 = triangle[0];
-    Point p2 = triangle[1];
-    Point p3 = triangle[2];
-
-    double center_x = (p1.x() + p2.x() + p3.x()) / 3;
-    double center_y = (p1.y() + p2.y() + p3.y()) / 3;
-
-    Point center = Point(center_x, center_y);
-
-    return center;
-}
-
-double Evaluation::getTriangleArea(LegacyTriangle triangle)
-{
-    Point p1 = triangle[0];
-    Point p2 = triangle[1];
-    Point p3 = triangle[2];
-
-    double area = abs(0.5 * ((p2.x() - p1.x()) * (p3.y() - p1.y()) -
-                             (p3.x() - p1.x()) * (p2.y() - p1.y())));
-
-    return area;
-}
-
-std::vector<LegacyTriangle> Evaluation::removeTrianglesOutsideRectangle(
-    Rectangle rectangle, std::vector<LegacyTriangle> triangles)
-{
-    std::vector<LegacyTriangle> valid_triangles;
-    std::vector<Point> rectangle_corners = {rectangle[0], rectangle[1], rectangle[2],
-                                            rectangle[3]};
+    std::vector<Triangle> valid_triangles;
     Point center;
 
-    double smallest_x = rectangle_corners[0].x();
-    double smallest_y = rectangle_corners[0].y();
-    double largest_x  = rectangle_corners[0].x();
-    double largest_y  = rectangle_corners[0].y();
-
-    for (unsigned int i = 0; i < rectangle_corners.size(); i++)
-    {
-        Point p = rectangle_corners[i];
-        if (p.x() < smallest_x)
-            smallest_x = p.x();
-        if (p.x() > largest_x)
-            largest_x = p.x();
-        if (p.y() < smallest_y)
-            smallest_y = p.y();
-        if (p.y() > largest_y)
-            largest_y = p.y();
-    }
+    double smallest_x = rectangle.negXNegYCorner().x();
+    double smallest_y = rectangle.negXNegYCorner().y();
+    double largest_x  = rectangle.posXPosYCorner().x();
+    double largest_y  = rectangle.posXPosYCorner().y();
 
     for (unsigned int i = 0; i < triangles.size(); i++)
```

Comment:
Out of scope, but I'd appreciate replacing this with `for each Triangle`
```suggestion
    for (Triangle t : triangles)
```

Reply:
Fixed in latest update